### DISCUSSION TITLE
Relationship Picker Filter not filtering

### DISCUSSION THREAD

**Discussion by koneubauer** [Budibase/budibase#12250] [Help] [answered]
I have a table **areas** and a table **zones**. I set up a relationship where one area can have many zones.
In the form I created from the **area** table, I have a Relationship Picker to select multiple zones for new and existing areas. I want the user to see only their zones in the picker. The zones have 5-digit identifiers. The first 3 digits correlate to the district. Users also have a attribute called district which is this 3-digit number. I set a filter: _zone-identifier_    _starts with_    _user-district-attribute_.
However, users are still shown all available zones. Even those, that dont start with their district-number.
Things to note: all districts, and therefore all zones start with 5

**Comment by ConorWebb96** [comment]
Hey @koneubauer,

How have you setup your relationship picker plus filters?

Could you provide some screenshots on this.

**Reply by koneubauer** [reply]
Hey,

I ejected the autogenerated block (lev3 is the name of the zones i mentioned:
![image](https://github.com/Budibase/budibase/assets/117980610/8f600d72-c33a-47fe-b536-010e8c86a40e)

And i set up the filter like this (name is the name of the zone and Current User.dgroup is the district-attribute of the user):
![image](https://github.com/Budibase/budibase/assets/117980610/dcdc2885-8e29-4bf3-a67c-c861d4855c38)

**Reply by ConorWebb96** [reply]
Hey @koneubauer,

- What does the binding `{{ Current User.dgroup }}` output?
- I'm not seeing how the relationship picker falls into what you have shown for your filters?

**Reply by koneubauer** [reply]
Hey,
- {{ Current User.dgroup }} outputs the district number. E.g. 506. The zones identifier could then be 50610
- I don' tfully understand your second question. Could you elaborate?

**Reply by ConorWebb96** [reply]
You are using `{{ Current User.dgroup }}` this is not the value of your relationship field shown in your screenshot.

I actually can't see any issues with this. I've tried this on my end and everything seems to filter fine? (I'm not filtering it in the exact same way but the output 506 is the same. It will give me results using `starts with` on the name field.) 

I'm on the latest version of Budibase.
<img width="1508" alt="Screenshot 2023-11-06 at 07 48 35" src="https://github.com/Budibase/budibase/assets/126772285/2b5be71d-48ee-439b-b848-34de9c71b821">

Working gif attached below;
![Screen Recording 2023-11-06 at 07 49 08](https://github.com/Budibase/budibase/assets/126772285/7cafa9c7-399c-49fd-858b-1ca39e74fb4b)

Is there a specific external db you are using? I'll try testing it with that as well.

**Reply by koneubauer** [reply]
Hey,
we're using an MS-SQL Server as the datasource. I'm also on the newest version. Maybe the issue is with the relationship picker?

**Reply by ConorWebb96** [reply]
In the example you sent, you weren't using the relationship to filter on the data provider? So I'm not sure what you are referring to.

I made an open issue about the relationship picker last week. So maybe this is what you [mean](https://github.com/Budibase/budibase/issues/12260)?

**Reply by koneubauer** [reply]
No, but I was using the User-attributes to filter the relationship picker. The second image is the filter of the relationship picker itself.

**Reply by ConorWebb96** [reply, marked as answer]
Ahh right, I understand you now, wasn't understanding how the relationship component was tying into any of this. 

I can confirm that the filtering on the relationship picker component doesn't seem to be working at all. I've gone ahead and created a bug report [here](https://github.com/Budibase/budibase/issues/12285). I can't think of any work arounds at the moment that could fix this. Therefore its likely going to need to be fixed in the core product.

Hopefully this helps.

**Reply by koneubauer** [reply]
Right, thanks. That explains it. Thanks for your help!

**Comment by immohacker** [comment]
Hi ConorWebb96,
thank you for change the prio but without this filter, my app is broken.
I think i revert to a an dataprovider with an option picker component, so i filter in the dataprovider but filter in relationship picker is very very nice. It makes things very very easy

I dont know which version it was 2.13.6 .213.7 2.13.8 but yesterday it would work short until i update today to latest (2.13.9).
I revert to every version list above but it would not work again.
I hope you can speed up the bug fix.
Thank you very much!
Best regards Bonkersdeluxe

**Comment by immohacker** [comment]
HiHo,
It works again @ Version 2.13.19
Thank you very much!
Best regards Bonkersdeluxe
